src/antsibull_changelog/changes.py

```diff
@@ -416,7 +421,17 @@ def update_fragments(
     def sort(self) -> None:
         """
         Sort change metadata in place.
-        """
+        Optionally sorts by semantic version.
+        """
+        if self.config.changelog_semantic_versioning_sort:
+            self.data["releases"] = dict(
+                sorted(
+                    self.data["releases"].items(),
+                    key=lambda t: [int(v) for v in t[0].split(".")],
+                    reverse=True,
```

Comment:
Doesn't that result in the reverse of the order you specified in the PR's description?

You really should add tests for this.

Reply:
Yes it ends with the reverse order to comply with the actual CHANGELOG.rst where the newest version is at the top of the file.
What I meant to highlight in the PR description was the placement of `0.2.0`, `0.2.1`, etc. which is incorrectly placed after `0.10.0` currently, but should be placed before it.

Reply:
Hmm, I personally would prefer the other order (together with semver ordering), i.e. oldest version at the top, and new versions added at the bottom. (Which we already have right now until you start using prereleases or two-digit version numbers ;) )

Maybe the field shouldn't be a boolean, but a string with four choices that give you the current order, the current order reversed, a version order, and a version order reversed.

Reply:
I actually don't have any preference on reverse or not myself, I simply thought reverse was preferred since both the generated `CHANGELOG.rst` and the `changelog.yaml` [antsichaut](https://github.com/ansible-community/antsichaut) generates both have the reversed order.

I considered adding an option to reverse the order, but it didn't seem to justify a whole new option key. Making it a string with four choices would be much better.

Reply:
For the generated files the order makes sense, since they are consumed by humans. For the YAML file, it can be whatever folks prefer... In case of antsichaud, I guess Brad introduced it in https://github.com/ansible-community/antsichaut/pull/35 when sorting by semver.